benchmarks/results/redis.md

```diff
@@ -0,0 +1,118 @@
+RedisListsBenchmark.redis4catsListWriter         thrpt    5   206.256 ± 555.952  ops/s
+```
+
+*Sets*
+```
+Benchmark                                       Mode  Cnt     Score     Error  Units
+RedisSetsBenchmark.laserdiscSetCardReader      thrpt    5   196.554 ±  36.414  ops/s
+RedisSetsBenchmark.laserdiscSetDiffWriter      thrpt    5   183.157 ±  63.127  ops/s
+RedisSetsBenchmark.laserdiscSetMembersReader   thrpt    5   181.293 ±  30.221  ops/s
+RedisSetsBenchmark.laserdiscSetWriter          thrpt    5   183.056 ±  37.632  ops/s
+RedisSetsBenchmark.redicolousSetCardReader     thrpt    5   173.508 ±  48.081  ops/s
+RedisSetsBenchmark.redicolousSetDiffWriter     thrpt    5   172.539 ±  27.593  ops/s
+RedisSetsBenchmark.redicolousSetMembersReader  thrpt    5   177.507 ±  24.302  ops/s
+RedisSetsBenchmark.redicolousSetWriter         thrpt    5   178.137 ±  45.988  ops/s
+RedisSetsBenchmark.redis4catsSetCardReader     thrpt    5   257.881 ±  68.513  ops/s
+RedisSetsBenchmark.redis4catsSetDiffWriter     thrpt    5   252.935 ± 119.800  ops/s
+RedisSetsBenchmark.redis4catsSetMembersReader  thrpt    5   254.934 ± 133.798  ops/s
+RedisSetsBenchmark.redis4catsSetWriter         thrpt    5   249.977 ± 136.127  ops/s
+RedisSetsBenchmark.setCardReader               thrpt    5  1735.714 ± 358.915  ops/s
+RedisSetsBenchmark.setDiffWriter               thrpt    5  1719.237 ± 224.646  ops/s
+RedisSetsBenchmark.setMembersReader            thrpt    5  1768.035 ± 249.180  ops/s
+RedisSetsBenchmark.setWriter                   thrpt    5  1707.503 ± 407.231  ops/s
+```
+
+*SortedSets*
+```
+Benchmark                                                 Mode  Cnt     Score     Error  Units
+RedisSortedSetsBenchmark.laserdiscSortedSetCardReader    thrpt    5   170.267 ±  89.732  ops/s
+RedisSortedSetsBenchmark.laserdiscSortedSetCountReader   thrpt    5   179.003 ±  40.080  ops/s
+RedisSortedSetsBenchmark.laserdiscSortedSetRangeReader   thrpt    5   178.721 ±  50.206  ops/s
+RedisSortedSetsBenchmark.laserdiscSortedSetWriter        thrpt    5   172.339 ±  30.774  ops/s
+RedisSortedSetsBenchmark.redicolousSortedSetCardReader   thrpt    5   158.050 ±  28.218  ops/s
+RedisSortedSetsBenchmark.redicolousSortedSetCountReader  thrpt    5   154.691 ±  28.389  ops/s
+RedisSortedSetsBenchmark.redicolousSortedSetRangeReader  thrpt    5   156.128 ±  22.739  ops/s
+RedisSortedSetsBenchmark.redicolousSortedSetWriter       thrpt    5   154.230 ±  38.051  ops/s
+RedisSortedSetsBenchmark.redis4catsSortedSetCardReader   thrpt    5   243.307 ±  60.217  ops/s
+RedisSortedSetsBenchmark.redis4catsSortedSetCountReader  thrpt    5   232.416 ±  66.536  ops/s
+RedisSortedSetsBenchmark.redis4catsSortedSetRangeReader  thrpt    5   236.565 ±  63.771  ops/s
+RedisSortedSetsBenchmark.redis4catsSortedSetWriter       thrpt    5   232.165 ±  54.573  ops/s
+RedisSortedSetsBenchmark.sortedSetCardReader             thrpt    5  1742.036 ± 234.697  ops/s
```

Comment:
@Avasil any idea why the `monix-redis` scores look so different to the rest of clients?  Not sure if that's a good sign or a bad one.

Reply:
I assume, that it could be because of other approach in using connection at other clients. Looks like they create new connection for every query

Reply:
No clue, I don't have a lot of experience benchmarking database connectors 😅 

Maybe we could compare it to the "raw" library that we are wrapping to see if it's similar to our results?
I assume that some of the other libraries are wrapping it too, the difference would be either:
1) They do too much in the implementation
2) We don't benchmark the same thing

I wouldn't expect big differences unless there are any streaming scenarios but it's mostly just Redis operations wrapped in `Task` / `F`, right?

Reply:
Yes we are just doing a simple wrapping over _lettuce_ library, which uses _Reactor_.
I believe that at least _redis4cats_ was also based on lettuce

A bit strange... but good to know that it is overperforming.